plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -163,16 +164,19 @@ def _get_camera_info(self, camera):
                               camera['signalStrength'])
 
     def _get_camera_snapshot(self, device_id):
-        if REAL_SETTINGS.getSetting('show_snapshots'):
+        if REAL_SETTINGS.getSetting('show_snapshots') == "true":
+            snapshot_path = "{0}/resources/media".format(ADDON_PATH)
```

Comment:
I've overlooked this last time, `ADDON_PATH` has to be `REAL_SETTINGS.getAddonInfo('profile').decode('utf-8')`. It isn't allowed to write into the add-on source folder. Also the files will be lost after each update.

Reply:
Not sure about the ADDON_PATH comment.  When I change it as you suggested to profile, I get an exception:
**Works:**
   def _get_camera_snapshot(self, device_id):
        if REAL_SETTINGS.getSettingBool('show_snapshots'):
            **snapshot_path = "{0}/resources/media".format(ADDON_PATH)**
            self.log("Snapshot location: {}".format(snapshot_path), xbmc.LOGNOTICE)
 
21:35:23.252 T:2108  NOTICE: [plugin.video.arloview-1.0.1] Snapshot location: C:\Users\a_dam\AppData\Roaming\Kodi\addons\plugin.video.arloview/resources/media

**Fails:**
    def _get_camera_snapshot(self, device_id):
        if REAL_SETTINGS.getSettingBool('show_snapshots'):
            **snapshot_path = "{0}/resources/media".format(SETTINGS_LOC**)
            self.log("Snapshot location: {}".format(snapshot_path), xbmc.LOGNOTICE)
            
21:41:53.887 T:19744  NOTICE: [plugin.video.arloview-1.0.1] Snapshot location: special://profile/addon_data/plugin.video.arloview//resources/media
21:41:53.891 T:19744   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.WindowsError'>
                                            Error Contents: [Error 123] The filename, directory name, or volume label syntax is incorrect: 'special:'
                                            Traceback (most recent call last):
                                              File "C:\Users\a_dam\AppData\Roaming\Kodi\addons\plugin.video.arloview\default.py", line 11, in <module>
                                                arlo_stream.ArloStream(sys.argv).run()
                                              File "C:\Users\a_dam\AppData\Roaming\Kodi\addons\plugin.video.arloview\resources\lib\arlo_stream.py", line 275, in run
                                                self.main_menu()
                                              File "C:\Users\a_dam\AppData\Roaming\Kodi\addons\plugin.video.arloview\resources\lib\arlo_stream.py", line 92, in main_menu
                                                snapshot_file = self._get_camera_snapshot(camera['deviceId'])
                                              File "C:\Users\a_dam\AppData\Roaming\Kodi\addons\plugin.video.arloview\resources\lib\arlo_stream.py", line 173, in _get_camera_snapshot
                                                os.makedirs(snapshot_path)